pyproject.toml

```
@@ -1,2 +1,3 @@
 [tool.ruff]
+[tool.ruff.lint]
 ignore = ["E501", "E731", "F841", "E721"]
```

I would suggest configuring Ruff to prefer single quote style since I believe it's more widely used in this project.

```suggestion
[tool.ruff.lint]
ignore = ["E501", "E731", "F841", "E721"]
[tool.ruff.format]
quote-style = "single"
```

Also, in another PR I guess, it would be nice to remove all unused variables by removing rule F841 from ignore list.

---

I would rather we move everything to what is most stanbdard including using double quotes.

I agre with removing the lint restrictions in separate PR's following this one.

---

Basically unless there is a very good reason not to use default options we should not customise linting and formatting. Generally the linting and formatting tools tend to reflect what is most standard in the industry and it will promote conformance across open source projects.